appservice/src/createAppService/AppInsightsCreateStep.ts

```diff
@@ -0,0 +1,58 @@
+import { Provider, ProviderResourceType } from 'azure-arm-resource/lib/resource/models';
+import { Location } from 'azure-arm-resource/lib/subscription/models';
+import { Progress } from 'vscode';
+import { AzureWizardExecuteStep, createAzureClient } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { nonNullProp, nonNullValue } from '../utils/nonNull';
+import { IAppServiceWizardContext } from './IAppServiceWizardContext';
+
+export class AppInsightsCreateStep extends AzureWizardExecuteStep<IAppServiceWizardContext> {
+    public priority: number = 135;
+
+    public async execute(wizardContext: IAppServiceWizardContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+        const location: Location = nonNullProp(wizardContext, 'location');
+        const verifyingAppInsightsAvailable: string = localize('verifyingAppInsightsAvailable', 'Verifying that application insights is available for this location...');
+        ext.outputChannel.appendLine(verifyingAppInsightsAvailable);
+        if (await this.appInsightsSupportedInLocation(wizardContext, location)) {
+
+            const creatingNewAppInsights: string = localize('creatingNewAppInsightsInsights', 'Creating new application insights component "{0}"...', wizardContext.newSiteName);
+            ext.outputChannel.appendLine(creatingNewAppInsights);
+            progress.report({ message: creatingNewAppInsights });
+
+            const client: ApplicationInsightsManagementClient = createAzureClient(wizardContext, ApplicationInsightsManagementClient);
+            wizardContext.applicationInsights = await client.components.createOrUpdate(
+                nonNullValue(wizardContext.newResourceGroupName),
+                nonNullValue(wizardContext.newApplicationInsightsName),
+                { kind: 'web', applicationType: 'web', location: nonNullProp(location, 'name') });
+            const createdNewAppInsights: string = localize('createdNewAppInsights', 'Created new application insights component "{0}"...', wizardContext.newSiteName);
+            wizardContext.aiInstrumentationKey = wizardContext.applicationInsights.instrumentationKey;
+            ext.outputChannel.appendLine(createdNewAppInsights);
+        } else {
+            const appInsightsNotAvailable: string = localize('appInsightsNotAvailable', 'Skipping creating an application insights component because it isn\'t compatible with this location.');
+            ext.outputChannel.appendLine(appInsightsNotAvailable);
+        }
+    }
+
+    public shouldExecute(wizardContext: IAppServiceWizardContext): boolean {
+        return !wizardContext.applicationInsights;
+    }
+
+    private async appInsightsSupportedInLocation(wizardContext: IAppServiceWizardContext, location: Location): Promise<boolean> {
+        const resourceClient: ResourceManagementClient = createAzureClient(wizardContext, ResourceManagementClient);
```

Comment:
ping on this

Reply:
Renamed to not be using the mapping.  I think we discussed offline that we would not use the mapping initially and add it in later (if there's a need for it)?

Reply:
I don't remember that discussion, but I'm okay if you want to do it separately. I think you should verify how many regions support function apps vs how many support app insights rather than waiting for user feedback. If there's a big gap there we should use the api ASAP

Reply:
![image](https://user-images.githubusercontent.com/5290572/62658294-52737800-b91d-11e9-8a12-cb717d756df9.png)

So for Functions, it seems like there's a 1:1 mapping.  But when I implement it for web apps, I'll probably have to use the mapping.

Reply:
You should check Windows, though. I don't think linux is a good sample because it's still in preview and the regions will likely increase as they GA